tests/subsys/bluetooth/mesh/scheduler_model/action_planning/src/main.c

```diff
@@ -0,0 +1,603 @@
+	},
+	{
+		.user_data = &scene_srv[2].srv,
+		.id = BT_MESH_MODEL_ID_SCENE_SRV,
+	},
+};
+
+static struct bt_mesh_model mock_models_elem4[3] = {
+	{
+		.user_data = &sched_srv[1],
+		.id = BT_MESH_MODEL_ID_SCHEDULER_SRV,
+	},
+	{
+		.user_data = &onoff_srv[3].srv,
+		.id = BT_MESH_MODEL_ID_GEN_ONOFF_SRV,
+	},
+	{
+		.user_data = &scene_srv[3].srv,
+		.id = BT_MESH_MODEL_ID_SCENE_SRV,
+	},
+};
+
+static struct bt_mesh_model mock_models_elem5[2] = {
+	{
+		.user_data = &onoff_srv[4].srv,
+		.id = BT_MESH_MODEL_ID_GEN_ONOFF_SRV,
+	},
+	{
+		.user_data = &scene_srv[4].srv,
+		.id = BT_MESH_MODEL_ID_SCENE_SRV,
+	},
+};
+
+static struct bt_mesh_elem mock_elems[5] = {
+	{ .addr = 1, .model_count = 3, .models = mock_models_elem1 },
+	{ .addr = 2, .model_count = 2, .models = mock_models_elem2 },
+	{ .addr = 3, .model_count = 2, .models = mock_models_elem3 },
+	{ .addr = 4, .model_count = 3, .models = mock_models_elem4 },
+	{ .addr = 5, .model_count = 2, .models = mock_models_elem5 }
+};
+
+struct mock_comp_data {
+	size_t elem_count;
+	struct bt_mesh_elem *elem;
+} mock_comp = { .elem_count = 5, .elem = mock_elems };
+
+static struct bt_mesh_model *sched_mod_elem1 = &mock_models_elem1[0];
+static struct bt_mesh_model *sched_mod_elem4 = &mock_models_elem4[0];
+
+static void mod_elem_idx_prep(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(mock_elems); i++) {
+		for (size_t j = 0; j < mock_elems[i].model_count; j++) {
+			mock_elems[i].models[j].elem_idx = i;
+		}
+	}
+}
+
+/* Redefined mocks */
+
+struct bt_mesh_elem *bt_mesh_model_elem(struct bt_mesh_model *mod)
+{
+	return &mock_comp.elem[mod->elem_idx];
+}
+
+struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
+{
+	uint16_t index;
+
+	if (!BT_MESH_ADDR_IS_UNICAST(addr)) {
+		return NULL;
+	}
+
+	index = addr - mock_comp.elem[0].addr;
+	if (index >= mock_comp.elem_count) {
+		return NULL;
+	}
+	return &mock_comp.elem[index];
+}
+
+struct bt_mesh_model *bt_mesh_model_find(const struct bt_mesh_elem *elem, uint16_t id)
+{
+	uint8_t i;
+
+	for (i = 0U; i < elem->model_count; i++) {
+		if (elem->models[i].id == id) {
+			return &elem->models[i];
+		}
+	}
+	return NULL;
+}
+
+int bt_mesh_scene_srv_set(struct bt_mesh_scene_srv *srv, uint16_t scene,
+			  struct bt_mesh_model_transition *transition)
+{
+	struct sched_evt *evt = sched_evt_next();
+	struct test_scene_srv *ptr = CONTAINER_OF(srv, struct test_scene_srv, srv);
+
+	zassert_equal(evt->elem_idx, ptr->elem_idx, "Wrong sequence of scene events");
+	zassert_equal(BT_MESH_SCHEDULER_SCENE_RECALL, evt->evt, "Incorrect scene event");
+	zassert_equal(evt->transition_time, transition->time, "Unexpected transition time");
+	zassert_equal(evt->scene_number, scene, "Incorrect Scene number");
+
+	if (fired_tm) {
+		fired_tm[galloc_cnt] = *bt_mesh_time_srv_localtime(&time_srv, k_uptime_get());
+		galloc_cnt++;
+	}
+
+	if (is_sched_evt_empty()) {
+		k_sem_give(&action_fired);
+	}
+	return 0;
+}
+
+struct tm *bt_mesh_time_srv_localtime(struct bt_mesh_time_srv *srv,
```

Comment:
`bt_mesh_time_srv_localtime` and `bt_mesh_time_srv_mktime` uses test member parameter `start_tm`. Seems hard to make a solution where we move these to a common library without makin it unnecessarily complex as far as I can see. Since these are mocs the header declerations can not be tampered with TTBOMK.